File: Example-Mac/PINRemoteImage/DegradedViewController.m
//
//  DegradedViewController.m
//  PINRemoteImage
//
//  Created by Michael Schneider on 1/6/16.
//  Copyright © 2016 mischneider. All rights reserved.
//

#import "DegradedViewController.h"

#import <PINRemoteImage/PINImageView+PINRemoteImage.h>

@interface DegradedViewController ()
@property (weak) IBOutlet NSImageView *imageView;
@end

@implementation DegradedViewController

#pragma mark - Lifecycle

- (instancetype)init
{
    self = [super initWithNibName:NSStringFromClass(self.class) bundle:nil];
    if (self == nil) { return self; }
    return self;
}


#pragma mark - NSViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [[PINRemoteImageManager sharedImageManager] setShouldUpgradeLowQualityImages:YES completion:nil];
}


- (void)viewWillAppear
{
    [super viewWillAppear];
    
    [self.imageView pin_setImageFromURLs:@[[NSURL URLWithString:@"https://placekitten.com/101/101"],
                                           [NSURL URLWithString:@"https://placekitten.com/401/401"],
                                           [NSURL URLWithString:@"https://placekitten.com/801/801"]]];
}

@end

File: Examples/Swift-Example/Swift-Example/DegradedViewController.swift
//
//  DegradedViewController.swift
//  Swift-Example
//
//  Created by Marius Landwehr on 23.01.16.
//  Copyright © 2016 Marius Landwehr. All rights reserved.
//

import UIKit
import PINRemoteImage

class DegradedViewController: UIViewController {

    @IBOutlet weak var imageView: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        PINRemoteImageManager.shared().setShouldUpgradeLowQualityImages(true, completion: nil)
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        imageView.pin_setImage(from: [
            URL(string: "https://placekitten.com/101/101")!,
            URL(string: "https://placekitten.com/401/401")!,
            URL(string: "https://placekitten.com/801/801")!])
    }
}
